packages/flutter_tools/lib/src/commands/upgrade.dart

```diff
@@ -107,16 +107,15 @@ class UpgradeCommandRunner {
     @required bool testFlow,
     @required bool verifyOnly,
   }) async {
-    final String upstreamRevision = await fetchRemoteRevision();
-    if (flutterVersion.frameworkRevision == upstreamRevision) {
+    final FlutterVersion upstreamVersion = await fetchLatestVersion();
+    if (flutterVersion.frameworkRevision == upstreamVersion.frameworkRevision) {
       globals.printStatus('Flutter is already up to date on channel ${flutterVersion.channel}');
       globals.printStatus('$flutterVersion');
       return;
     } else if (verifyOnly) {
       globals.printStatus('A new version of Flutter is available on channel ${flutterVersion.channel}\n');
-      // TODO(fujino): use a [FlutterVersion] once that class supports arbitrary revisions.
-      globals.printStatus('The latest revision: $upstreamRevision', emphasis: true);
-      globals.printStatus('Your current version: ${flutterVersion.frameworkRevision}\n');
+      globals.printStatus('The latest version: ${upstreamVersion.frameworkVersion}(revision ${upstreamVersion.frameworkRevisionShort})', emphasis: true);
```

Comment:
Please add a space between the version and the parentheses here and on the next line. This currently looks like:

```
The latest version: 1.24.0-8.0.pre.5(revision 1f15046b93)
Your current version: 1.24.0-8.0.pre.6(revision bf0fc66cbf)
```

We should also ensure that in the unit tests we're asserting that this is formatted correctly.